src/core/project.js

```diff
@@ -1,166 +0,0 @@
-  +discord: Discord | null,
-  +identities: $ReadOnlyArray<Identity>,
-  +timelineCredParams: $Shape<TimelineCredParameters> | null,
-|};
-
-const COMPAT_INFO = {type: "sourcecred/project", version: "0.5.2"};
-
-/**
- * Creates a new Project instance with default values.
- *
- * Note: the `id` field is required, as there's no sensible default.
- */
-export function createProject(p: $Shape<Project>): Project {
-  if (!p.id) {
-    throw new Error("Project.id must be set");
-  }
-
-  return {
-    repoIds: [],
-    identities: [],
-    discourseServer: null,
-    initiatives: null,
-    timelineCredParams: null,
-    discord: null,
-    ...p,
-  };
-}
-
-export type ProjectJSON = Compatible<Project>;
-
-export function projectToJSON(p: Project): ProjectJSON {
-  return toCompat(COMPAT_INFO, p);
-}
-
-export function projectFromJSON(j: Compatible<any>): Project {
-  return fromCompat(COMPAT_INFO, j, upgrades);
-}
-
-/**
- * Encode the project ID so it can be stored on the filesystem,
- * or retrieved via XHR from the frontend.
- */
-export function encodeProjectId(id: ProjectId): string {
-  return base64url.encode(id);
-}
-
-const upgradeFrom051 = (p: ProjectV051): ProjectV052 => ({
-  discord: null,
-  ...p,
-});
-
-export type ProjectV051 = {|
-  +id: ProjectId,
-  +initiatives: Initiatives | null,
-  +repoIds: $ReadOnlyArray<RepoId>,
-  +discourseServer: DiscourseServer | null,
-  +identities: $ReadOnlyArray<Identity>,
-  +timelineCredParams: $Shape<TimelineCredParameters> | null,
-|};
-
-const upgradeFrom050 = (p: ProjectV050) =>
-  upgradeFrom051({
-    timelineCredParams: {},
-    ...p,
-  });
-
-export type ProjectV050 = {|
-  +id: ProjectId,
-  +initiatives: Initiatives | null,
-  +repoIds: $ReadOnlyArray<RepoId>,
-  +discourseServer: DiscourseServer | null,
-  +identities: $ReadOnlyArray<Identity>,
-|};
-
-const upgradeFrom040 = (p: ProjectV040) =>
-  upgradeFrom050({
-    initiatives: null,
-    ...p,
-  });
-
-export type ProjectV040 = {|
-  +id: ProjectId,
-  +repoIds: $ReadOnlyArray<RepoId>,
-  +discourseServer: DiscourseServer | null,
-  +identities: $ReadOnlyArray<Identity>,
-|};
-
-const upgradeFrom030 = (p: ProjectV030 | ProjectV031) =>
-  upgradeFrom040({
-    ...p,
-    discourseServer:
-      p.discourseServer != null
-        ? {serverUrl: p.discourseServer.serverUrl}
-        : null,
-  });
-
-export type ProjectV031 = {|
-  +id: ProjectId,
-  +repoIds: $ReadOnlyArray<RepoId>,
-  +discourseServer: {|
-    +serverUrl: string,
-    +apiUsername?: string,
-  |} | null,
-  +identities: $ReadOnlyArray<Identity>,
-|};
-
-export type ProjectV030 = {|
-  +id: ProjectId,
-  +repoIds: $ReadOnlyArray<RepoId>,
-  +discourseServer: {|
-    +serverUrl: string,
-    +apiUsername: string,
-  |} | null,
-  +identities: $ReadOnlyArray<Identity>,
-|};
-
-const upgrades = {
-  "0.3.0": upgradeFrom030,
-  "0.3.1": upgradeFrom030,
-  "0.4.0": upgradeFrom040,
-  "0.5.0": upgradeFrom050,
-  "0.5.1": upgradeFrom051,
```

Comment:
Do we no longer want to have any sort of upgrade path from the legacy project.json to the newer format? At least something that will generate an error message letting the user know how they can migrate their config?

Reply:
The new setup is really different--IMO users are more likely to want to start fresh with a fork off the template repo. Trying to automate upgrades would be a lot of effort relative to our small number of users IMO.